ext/RMagick/rmmain.c

```diff
@@ -98,6 +110,75 @@ static void rm_free(void *ptr)
 }
 
 
+
+#if USE_RM_ALIGNED_MALLOC
+
+static size_t
+rm_aligned_malloc_size(void *ptr)
+{
+#if defined(HAVE_MALLOC_USABLE_SIZE)
+    return malloc_usable_size(ptr);
+#elif defined(HAVE_MALLOC_SIZE)
+    return malloc_size(ptr);
+#elif defined(_WIN32)
+    return _aligned_msize(ptr, 0, 0);
+#endif
+}
+
+
+/**
+ * Allocate aligned memory.
+ *
+ * No Ruby usage (internal function)
+ *
+ * @param size the size of memory to allocate
+ * @return pointer to a block of memory
+ */
+static void *rm_aligned_malloc(size_t size, size_t alignment)
+{
+    void *res;
+
+#if defined(HAVE_POSIX_MEMALIGN)
+    if (posix_memalign(&res, alignment, size) != 0) {
+        return NULL;
+    }
+#elif defined(HAVE_MEMALIGN)
+    res = memalign(alignment, size);
+#elif defined(_WIN32)
+    res = _aligned_malloc(size, alignment);
+#endif
+
+    size = rm_aligned_malloc_size(res);
+    rb_gc_adjust_memory_usage(size);
+    return res;
+}
+
+
+
+
+/**
+ * Free aligned memory.
+ *
+ * No Ruby usage (internal function)
+ *
+ * @param ptr pointer to the existing block of memory
+ */
+static void rm_aligned_free(void *ptr)
+{
+    size_t size = rm_aligned_malloc_size(ptr);
```

Comment:
It looks like we would also need `alignment` here but I don't have an idea where that should be stored then. We have a fixed value in our header file:

`memory-private.h:`
```
#if defined(__powerpc__)
#  define CACHE_LINE_SIZE  (16 * MAGICKCORE_SIZEOF_VOID_P)
#else
#  define CACHE_LINE_SIZE  (8 * MAGICKCORE_SIZEOF_VOID_P)
#endif
```

```
  if (memory_methods.acquire_aligned_memory_handler != (AcquireAlignedMemoryHandler) NULL)
    return(memory_methods.acquire_aligned_memory_handler(size,CACHE_LINE_SIZE));
```

So maybe we could just use that instead?

Reply:
https://github.com/rmagick/rmagick/pull/836/commits/8f0ef8350c1435988b9cd576aca50ce4e15b4115